pkg/controllers/provisioningv2/rke2/provisioningcluster/template.go

```diff
@@ -229,28 +238,47 @@ func createMachineTemplateHash(dataMap map[string]interface{}) string {
 }
 
 func machineDeployments(cluster *rancherv1.Cluster, capiCluster *capi.Cluster, dynamic *dynamic.Controller,
-	dynamicSchema mgmtcontroller.DynamicSchemaCache, secrets v1.SecretCache) (result []runtime.Object, _ error) {
+	dynamicSchema mgmtcontroller.DynamicSchemaCache, secrets v1.SecretCache, capiMachineDeployments capicontrollers.MachineDeploymentCache, rkeBootstrapTemplates rkev1controllers.RKEBootstrapTemplateCache) (result []runtime.Object, _ error) {
 	bootstrapName := name.SafeConcatName(cluster.Name, "bootstrap", "template")
 
 	if dynamicSchema == nil {
 		return nil, nil
 	}
 
 	if len(cluster.Spec.RKEConfig.MachinePools) > 0 {
-		result = append(result, &rkev1.RKEBootstrapTemplate{
-			ObjectMeta: metav1.ObjectMeta{
-				Namespace: cluster.Namespace,
-				Name:      bootstrapName,
-			},
-			Spec: rkev1.RKEBootstrapTemplateSpec{
-				ClusterName: cluster.Name,
-				Template: rkev1.RKEBootstrap{
-					Spec: rkev1.RKEBootstrapSpec{
-						ClusterName: cluster.Name,
+		ls, err := labels.Parse(fmt.Sprintf("%s=%s", rke2.ClusterNameLabel, cluster.Name))
+		if err != nil {
+			return nil, err
+		}
+		rbt, err := rkeBootstrapTemplates.List(cluster.Namespace, ls)
+		if err != nil {
+			return nil, err
+		}
+		if len(rbt) > 1 {
+			return nil, fmt.Errorf("multiple rkebootstraptemplates found for cluster %s/%s", cluster.Namespace, cluster.Name)
```

Comment:
Currently, there is one bootstrap template per machine pool. For a cluster with multiple machine pools this will always fail as there will be multiple rkebootstraptemplates present by design.

Reply:
Resolving this as the change to move to one bootstrap template per machine pool was not added